### ISSUE TITLE
[FR]: Router being blocked not end client IP address

### ISSUE THREAD

**Issue by atcapollo** [fail2ban/fail2ban#2970]
I have the filter for home-assistant 'hass-iptables'. It correctly identifies when to ban, but it bans the router 192.168.1.1. Not the client.
The log files suggests it uses fail2ban.ipdns, where as my other bans e.g. for Owncloud / Proftpd use fail2ban.actions and correctly ban the client IP.

Why does the hass-iptables jail use fail2ban.ipdns?
How can I get it to ban the client not the router?

Thanks for any pointers, this is a great tool.

### Environment:

Fail2Ban version:
fail2ban:
Installed: 0.11.1-1
Candidate: 0.11.1-1
Version table:
*** 0.11.1-1 500
500 http://gb.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
500 http://gb.archive.ubuntu.com/ubuntu focal/universe i386 Packages

OS, including release name/version :
LSB Version: core-11.1.0ubuntu2-noarch:security-11.1.0ubuntu2-noarch
Distributor ID: Ubuntu
Description: Ubuntu 20.04.1 LTS
Release: 20.04
Codename: focal

#### Service, project or product which log or journal should be monitored

- Name of filter or jail in Fail2Ban (if already exists) : hass-iptables
- Service, project or product name, including release name/version : Home Assistant 
- Repository or URL (if known) : 
- Service type :  systemd
- Ports and protocols the service is listening :  8123

Jail.local
```
[hass-iptables]
enabled = true
filter = hass
action = iptables-allports[name=HASS]
logpath = /home/homeassistant/.homeassistant/home-assistant.log
#OLD logpath = /var/log/homeassistant/home-assistant.log
maxretry = 3
bantime = 3600
```

filter.d/hass.conf

```
INCLUDES]
before = common.conf

[Definition]
failregex = ^%(__prefix_line)s.*Login attempt or request with invalid authentication from <HOST>.*$

ignoreregex =

[Init]
datepattern = ^%%Y-%%m-%%d %%H:%%M:%%S
```
#### Log or journal information

- Log file name(s) : 
/var/log/fail2ban.log

Relevant lines:
```
2021-03-12 23:46:58,153 fail2ban.filter         [416297]: INFO    [hass-iptables] Found 192.168.1.1 - 2021-03-12 23:44:00
2021-03-12 23:46:58,153 fail2ban.ipdns          [416297]: WARNING Determined IP using DNS Lookup: House.london = {'192.168.1.1'}
```

Complete tail:
```
2021-03-12 23:46:58,152 fail2ban.filter         [416297]: INFO    [hass-iptables] Found 192.168.1.1 - 2021-03-12 23:43:59
2021-03-12 23:46:58,153 fail2ban.ipdns          [416297]: WARNING Determined IP using DNS Lookup: House.london = {'192.168.1.1'}
2021-03-12 23:46:58,153 fail2ban.filter         [416297]: INFO    [hass-iptables] Found 192.168.1.1 - 2021-03-12 23:44:00
2021-03-12 23:46:58,153 fail2ban.ipdns          [416297]: WARNING Determined IP using DNS Lookup: House.london = {'192.168.1.1'}
2021-03-12 23:46:58,154 fail2ban.filter         [416297]: INFO    [hass-iptables] Found 192.168.1.1 - 2021-03-12 23:46:58
2021-03-12 23:46:58,422 fail2ban.actions        [416297]: NOTICE  [hass-iptables] Ban 192.168.1.1
2021-03-12 23:48:08,061 fail2ban.filter         [416297]: INFO    [proftpd] Found 192.168.1.146 - 2021-03-12 23:48:08
2021-03-12 23:48:18,982 fail2ban.filter         [416297]: INFO    [proftpd] Found 192.168.1.146 - 2021-03-12 23:48:18
2021-03-12 23:48:28,005 fail2ban.filter         [416297]: INFO    [proftpd] Found 192.168.1.146 - 2021-03-12 23:48:27
2021-03-12 23:48:28,517 fail2ban.actions        [416297]: NOTICE  [proftpd] Ban 192.168.1.146


<!-- Systemd journal -->

- Journal identifier or unit name : 
hass-iptables

#### Any additional information
n/a

### Relevant lines from monitored log files:


#### failures in sense of fail2ban filter (fail2ban must match):
<!-- put your log excerpt between next 2 lines -->
```
```

#### legitimate messages (fail2ban should not consider as failures):
<!-- put your log excerpt between next 2 lines -->
```
```

**Comment by sebres** [comment]
You did not provide the excerpt of counterpart log of home assistant (e. g. failure found at 2021-03-12 23:46:58).

I guess it will be something like:
```
... Login attempt or request with invalid authentication from House.london
... Login attempt or request with invalid authentication from 192.168.1.1
```

1. If it is either the first, then for some reason `House.london` resolving `192.168.1.1`.
2. If it is the second, then you get in log only IP address of the router.

To avoid issue in first case, you can either disable dns-lookup (`usedns = no`) or replace `<HOST>` with `<ADDR>`. Just note that fail2ban will expect only IP address in the log, but it is always good to do that, if your intention is to ban IP addresses only (or the log must basically contain IPs only). 

In any case to stop banning of your router (or some other IPs or hosts) you can also specify its address or name in `ignoreip`.
For example `ignoreip = 192.168.1.1/24` will completely avoid ban for your local network addresses.

But I guess your real issue is not this, but rather that home assistant does not log the real IP address or your router does not forward it correctly to the host (rpi or whatever where the HA is running). Did you ever see some foreign IP address after `from` in the home-assistant.log?
If not, this is the matter of your router configuration: whether or how it forwards the external traffic to the HA-host or how it is proxying such requests (and then whether HA follows XFF (`X-Forwared-For`) or other headers your router may send in a proxy request).

**Comment by sebres** [comment]
Yep, looks like either your configuration (router forwarding rules?) or the tests are not correct:

If I try to connect from my smartphone to my HA behind router port forwarding with wrong password, I see `invalid authentication from 89.15.X.X`, which is my mobile IP address (if I try to access external address or disable WiFi on smartphone, otherwise I see 192.X.X.123 which is internal IP of my smartphone, but not the routers).

Oh wait! How you try to communicate with your HA? From computer withing your network? Then I can probably explain it:
- you use something like DynDNS/whatever else that makes possible that your home assistant host is accessible via something like `ha.your-name.example.com` which resolves external address of your router (for example `222.111.222.111`).
- you try to connect from your computer (behind your router) to `ha.your-name.example.com` (e. g. with browser)
- in this case the route for your packets from browser to HA looks like:
`192.168.1.146 -> 192.168.1.1 -> outside world -> [ 222.111.222.111 | 192.168.1.1 ] --forwards-to--> 192.168.1.5 (HA)`

So from point of view of your HA (Owncloud) the request was coming from 192.168.1.1.
Just try to connect HA-host using internal name (e. g. ha.name-of-your-router) or internal address (192.168.1.5) and it would find and ban the local client 192.168.1.146 exactly in the same way as proftpd.

But basically it is wrong strategy - I assume you don't want to ban the devices from your local network aka intrAnet, rather from outside (intErnet)...
For such a test you must either try from mobile devices connected via mobile internet (not using WiFi of your router) or from some vpn or external host.

**Comment by atcapollo** [comment]
Thanks for taking the time to clarify - indeed I was testing under the scenario you highlighted:

`
Queried domain.com resolved to 222.111.222.111
192.168.1.146 -> 192.168.1.1 -> outside world -> [ 222.111.222.111 | 192.168.1.1 ] --forwards-to--> 192.168.1.5 (HA)
`

That makes more sense. I shall continue to test. Sorry I may have wasted your time on this.
